readium/shared/src/main/java/org/readium/r2/shared/util/pdf/PdfDocument.kt

```diff
@@ -33,7 +33,7 @@ interface PdfDocumentFactory<T : PdfDocument> {
     suspend fun open(file: File, password: String?): T
 
     /** Opens a PDF from a Fetcher resource. */
-    suspend fun open(resource: Resource, password: String?): T
+    suspend fun open(resource: Fetcher.Resource, password: String?): T
```

Comment:
It feels to me that the `PdfDocument` should depend on the core `Resource` instead of the `Fetcher` one. I looked into the implementation and there's no use of the `link` that `Fetcher.Resource` adds, except in `CachingPdfDocumentFactory`. This one could use `name()` as an alternative, and not cache it if the `name()` is missing. Alternatively, we could have `open(name: String, resource: Resource)`.